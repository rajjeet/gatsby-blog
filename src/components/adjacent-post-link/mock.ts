export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
