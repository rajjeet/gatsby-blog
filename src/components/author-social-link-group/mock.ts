import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

