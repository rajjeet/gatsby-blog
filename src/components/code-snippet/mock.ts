import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
