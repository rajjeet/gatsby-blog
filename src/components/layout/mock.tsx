import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

