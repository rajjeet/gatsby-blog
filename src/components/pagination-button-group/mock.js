export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
