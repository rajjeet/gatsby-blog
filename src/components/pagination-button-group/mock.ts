export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
