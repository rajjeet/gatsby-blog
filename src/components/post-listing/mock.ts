import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

