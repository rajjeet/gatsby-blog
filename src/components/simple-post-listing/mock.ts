import { createMockPosts } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  posts: createMockPosts,
});

