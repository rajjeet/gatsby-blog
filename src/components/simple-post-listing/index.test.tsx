import React from 'react';
import { render } from '@testing-library/react';
import { SimplePostListing } from './index';
import { makeProps } from './mock';

describe('<SimplePostListing />', () => {
  it('should render', () => {
    const { asFragment } = render(<SimplePostListing {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a heading', () => {
    const { getByText } = render(<SimplePostListing {...makeProps()} />);
    expect(getByText('Blog Posts')).toBeDefined();
  });

  it('should a link for each listed post', () => {
    const { getByLabelText, queryAllByRole } = render(<SimplePostListing {...makeProps()} />);
    expect(queryAllByRole('link')).toHaveLength(makeProps().posts.length);
    expect(getByLabelText('See post: Adding a Project Section to My Website')).toBeDefined();
    expect(getByLabelText('See post: State Management using React Hooks')).toBeDefined();
  });

  it('should "time to read" metrics for each post', () => {
    const { getByText } = render(<SimplePostListing {...makeProps()} />);
    expect(getByText('2 mins read')).toBeDefined();
    expect(getByText('4 mins read')).toBeDefined();
  });

  it('should have the post tiles for each post', () => {
    const { getByText } = render(<SimplePostListing {...makeProps()} />);
    expect(getByText('Adding a Project Section to My Website')).toBeDefined();
    expect(getByText('State Management using React Hooks')).toBeDefined();
  });

  it('should have the post dates', () => {
    const { getByText } = render(<SimplePostListing {...makeProps()} />);
    expect(getByText('11 May, 2019')).toBeDefined();
    expect(getByText('02 May, 2019')).toBeDefined();
  });
});
