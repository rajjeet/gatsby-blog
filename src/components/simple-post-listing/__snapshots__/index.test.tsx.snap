// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimplePostListing /> should render 1`] = `
<DocumentFragment>
  .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c1 h4 {
  margin: 0 auto;
}

.c1 div {
  font-size: .8rem;
}

.c0 {
  list-style: none;
  margin-left: 0;
  padding-left: 0;
}

.c0 li {
  margin-bottom: .25rem;
  padding: .7rem;
}

.c0 li:hover {
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
  -webkit-transition: ease .3s;
  transition: ease .3s;
}

.c0 a {
  color: black;
}

.c2 {
  padding-right: .5rem;
}

<div>
    <h3>
      Blog Posts
    </h3>
    <ul
      class="c0"
    >
      <a
        aria-label="See post: Adding a Project Section to My Website"
        class="c1"
        href="/adding-project-section-to-website/"
      >
        <li>
          <h4>
            Adding a Project Section to My Website
          </h4>
          <div>
            <span
              class="c2"
            >
              11 May, 2019
            </span>
            <span>
              2 mins read
            </span>
          </div>
        </li>
      </a>
      <a
        aria-label="See post: State Management using React Hooks"
        class="c1"
        href="/state-management-react-hooks/"
      >
        <li>
          <h4>
            State Management using React Hooks
          </h4>
          <div>
            <span
              class="c2"
            >
              02 May, 2019
            </span>
            <span>
              4 mins read
            </span>
          </div>
        </li>
      </a>
      <a
        aria-label="See post: Build A Super Customized and Performant Blog"
        class="c1"
        href="/build-super-customized-performant-blog/"
      >
        <li>
          <h4>
            Build A Super Customized and Performant Blog
          </h4>
          <div>
            <span
              class="c2"
            >
              21 February, 2019
            </span>
            <span>
              3 mins read
            </span>
          </div>
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
